app/src/main/java/com/example/cookbook/ui/create_recipe/OwnRecipesViewModel.kt

```diff
@@ -32,26 +37,27 @@ class OwnRecipesViewModel @Inject constructor(private val recipeRepository: OwnR
     }
 
     private fun observeRecipes() {
-        _isLoading.value = true
-
-        val disposable = recipeRepository.getRecipeListSync()
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .subscribe {
-                ownRecipes.postValue(it)
-                _isLoading.value = false
-            }
-
-        composite.add(disposable)
+        recipeRepository.getRecipeListSync()
+            .subscribeOn(schedulerProvider.io())
+            .observeOn(schedulerProvider.mainThread())
+            .doOnSubscribe { _isLoading.value = true }
+            .subscribe({
+                _ownRecipes.value = it
+                _isLoading.value = ownRecipes.value?.isEmpty() == true
+            }, {
+                it.localizedMessage
```

Comment:
In case of error we should do nothing ? Let's add some Toast or Log here

Reply:
ok